chart/org.eclipse.birt.chart.ui.extension/src/org/eclipse/birt/chart/ui/swt/composites/ImageDialog.java

```diff
@@ -296,6 +309,40 @@ public void widgetSelected(SelectionEvent event) {
 		});
 	}
 
+	private void createImageBrowseButton(Composite innerComp) {
+		browseButton = new Button(innerComp, SWT.PUSH);
+		browseButton.setText(Messages.getString("ImageDialog.label.Browse")); //$NON-NLS-1$
+		browseButton.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_END));
+		browseButton.addSelectionListener(new SelectionAdapter() {
+
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				FileDialog fileChooser = new FileDialog(getShell(), SWT.OPEN);
+				fileChooser.setText(Messages.getString("ImageDialog.label.SelectFile")); //$NON-NLS-1$
+				fileChooser.setFilterExtensions(new String[] { "*.gif", "*.jpg", "*.png" //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+				});
+				try {
+					String fullPath = fileChooser.open();
+					if (fullPath != null) {
+						String fileName = fileChooser.getFileName();
+						if (fileName != null) {
+							imageData = null;
+							imageURI = fullPath;
+							if (selectType.equals(SelectType.EMBEDDED_TYPE)) {
+								uriEditor.setText(fileName);
```

Comment:
The place where you know it's a file name, I highly recommend properly converting it to a URI:
```suggestion
								uriEditor.setText(Path.of(fileName).toUri().toString());
```